cs/quickstart/refactoring-model.texy

```diff
@@ -0,0 +1,121 @@
+Model
+**************
+
+S tím, jak aplikace roste, brzy zjistíme, že na různých místech, v různých presenterech, potřebujeme provádět podobné operace s databází, například získávat nejnovější publikované články. Když aplikaci vylepšíme třeba tím, že u článků přidáme příznak, zda je rozepsaný, musíme potom projít i všechna místa v aplikaci, kde se články z databáze získávají, a doplnit podmínku where, aby se vybíraly jen články nerozepsané.
+
+V ten moment se přímá práce s databází stává nedostatečnou a bude šikovnější si vypomoci novou funkcí, která nám publikované články bude vracet. A když později přidáme další podmínku, například že se nemají zobrazovat články s budoucím datem, upravíme kód jen na jednom místě.
+
+Funkci umístíme třeba do třídy `ArticleManager` a nazveme ji `getPublicArticles()`.
+
+
+V adresáři `app/model/` vytvoříme naši modelovou třídu, která se nám bude starat o články, a pojmenujeme ji `ArticlesManager.php`.
+
+
+/---php
+<?php
+
+namespace App\Model;
+
+use Nette;
+
+/**
+ * Article management.
+ */
+class ArticleManager extends Nette\Object
+{
+	/**
+	 * @var Nette\Database\Context
+	 */
+	private $database;
+
+	public function __construct(Nette\Database\Context $database)
+	{
+		$this->database = $database;
+	}
+
+	public function getPublicArticles()
+	{
+		return $this->database->table('posts')
+			->where('created_at < ', new \DateTime())
+			->order('created_at DESC');
+	}
+}
+
+\--
+
+Třídu jsme vytvořili s konstruktorem, ve kterém si předáme databázový Context:[api:Nette\Database\Context] do managera.
+
+Pokud bychom nevyužívali frameworku a síly DI containeru, museli bychom v našem případě nejprve vytvořit instaci `Nette\Database\Connection`, `Nette\Database\Structure`, `Nette\Database\Conventions` a `Nette\Caching\Storages\FileStorage` (které pro své instancování také potřebují nějaké závislosti, třeba jméno a heslo k databázi) pro to, abychom mohli vytvořit instanci `Nette\Database\Context`, kterou bychom následně použili pro vytvoření instance `ArticleManager`.
+
+/--php
+	...
+	// vyrobime vse co je treba pro Context
```

Comment:
Toto jsem chápal jako úmysl kvůli tomu, že je to součástí kódu. Preferovanější je tedy diakritika?